### ISSUE TITLE
[Bug?] [Help request] v2.11.3无法使用本地的webdav服务

### ISSUE THREAD

**Issue by Wedjat98** [ChatGPTNextWeb/ChatGPT-Next-Web#4335]
### Bug Description

*No PROXY_URL confirmed*

我有一个极其简易的webdav服务如下：
```js
const { v2: webdav } = require('webdav-server');

// 创建一个用户管理器
const userManager = new webdav.SimpleUserManager();
const user = userManager.addUser('user', 'pass', false);

// 创建权限管理器并设置权限
const privilegeManager = new webdav.SimplePathPrivilegeManager();
privilegeManager.setRights(user, '/', ['all']);

// 创建WebDAV服务器实例
const server = new webdav.WebDAVServer({
    port: 3900,
    httpAuthentication: new webdav.HTTPBasicAuthentication(userManager),
    privilegeManager: privilegeManager
});

// 设置文件系统并添加错误处理
server.setFileSystem('/', new webdav.PhysicalFileSystem('someDir/webdav'), (success, error) => {
    if (error) {
        console.error('Error setting file system:', error);
        return;
    }
    console.log(`Filesystem loaded successfully: ${success}`);
});

// 在请求开始和结束时添加详细的日志记录
server.beforeRequest((req, next) => {
    console.log(`Incoming request: ${req.method} ${req.url}`);
    console.log(`Headers: ${JSON.stringify(req.headers)}`);
    next();
});

server.afterRequest((req, next) => {
    console.log(`Completed request: ${req.method} ${req.url} - Status: ${req.response.statusCode}`);
    next();
});

// 启动服务器
server.start(() => console.log('WebDAV server started on port 3900'));

```

我在v2.11.2 是可以正常使用这个服务来同步对话历史到不同的设备，但是升级至v2.11.3我无法使用了。请问我应该怎么做？或者我应该从哪里获取到同步失败的相关log？

### Steps to Reproduce

update to 2.11.3 and Sync the chat histories then show dialog `Sync Fail`

### Expected Behavior

`Sync Success`

### Deployment Method

- [ ] Docker
- [ ] Vercel
- [X] Server

**Comment by fred-bf** [comment]
@Wedjat98 could you share the console log from your browser when sync with webdav?

**Comment by Wedjat98** [comment]
@fred-bf 

Sure! Here is it.

```js
[Sync] failed to get remote state TypeError: URL constructor: /api/cors//api/webdav/chatgpt-next-web/backup.json/ is not a valid URL.
    NextJS 33
[page-606a277117982b1d.js:1:232420](http://192.168.0.244:3000/_next/static/chunks/app/page-606a277117982b1d.js)
[Sync] TypeError: URL constructor: /api/cors//api/webdav/chatgpt-next-web/backup.json/ is not a valid URL.
    NextJS 33
[7698-bb5d18468650f39a.js:1:3423](http://192.168.0.244:3000/_next/static/chunks/7698-bb5d18468650f39a.js)
    NextJS 31
```

**Comment by mosimod** [comment]
I have the save issue

**Comment by fred-bf** [comment]
working on it, it should cause by a recent security update

**Comment by yuluod** [comment]
> 我在使用坚果云webdav，点击“同步”显示成功了，但是去网盘实际看文件没有更新

同样也遇到了这个问题

**Comment by H0llyW00dzZ** [comment]
> ### Bug Description
> _No PROXY_URL confirmed_
> 
> 我有一个极其简易的webdav服务如下：
> 
> ```js
> const { v2: webdav } = require('webdav-server');
> 
> // 创建一个用户管理器
> const userManager = new webdav.SimpleUserManager();
> const user = userManager.addUser('user', 'pass', false);
> 
> // 创建权限管理器并设置权限
> const privilegeManager = new webdav.SimplePathPrivilegeManager();
> privilegeManager.setRights(user, '/', ['all']);
> 
> // 创建WebDAV服务器实例
> const server = new webdav.WebDAVServer({
>     port: 3900,
>     httpAuthentication: new webdav.HTTPBasicAuthentication(userManager),
>     privilegeManager: privilegeManager
> });
> 
> // 设置文件系统并添加错误处理
> server.setFileSystem('/', new webdav.PhysicalFileSystem('someDir/webdav'), (success, error) => {
>     if (error) {
>         console.error('Error setting file system:', error);
>         return;
>     }
>     console.log(`Filesystem loaded successfully: ${success}`);
> });
> 
> // 在请求开始和结束时添加详细的日志记录
> server.beforeRequest((req, next) => {
>     console.log(`Incoming request: ${req.method} ${req.url}`);
>     console.log(`Headers: ${JSON.stringify(req.headers)}`);
>     next();
> });
> 
> server.afterRequest((req, next) => {
>     console.log(`Completed request: ${req.method} ${req.url} - Status: ${req.response.statusCode}`);
>     next();
> });
> 
> // 启动服务器
> server.start(() => console.log('WebDAV server started on port 3900'));
> ```
> 
> 我在v2.11.2 是可以正常使用这个服务来同步对话历史到不同的设备，但是升级至v2.11.3我无法使用了。请问我应该怎么做？或者我应该从哪里获取到同步失败的相关log？
> 
> ### Steps to Reproduce
> update to 2.11.3 and Sync the chat histories then show dialog `Sync Fail`
> 
> ### Expected Behavior
> `Sync Success`
> 
> ### Screenshots
> _No response_
> 
> ### Deployment Method
> * [ ]  Docker
> * [ ]  Vercel
> * [x]  Server
> 
> ### Desktop OS
> _No response_
> 
> ### Desktop Browser
> _No response_
> 
> ### Desktop Browser Version
> _No response_
> 
> ### Smartphone Device
> _No response_
> 
> ### Smartphone OS
> _No response_
> 
> ### Smartphone Browser
> _No response_
> 
> ### Smartphone Browser Version
> _No response_
> 
> ### Additional Logs
> _No response_

try this #4353

**Comment by Wedjat98** [comment]
I have rolled back to the v2.11.2. 
When I clicked the Check Connection button, Console showed this:

[CORS] target =  /api/cors/http/192.168.0.244:3900/chatgpt-next-web [page-8ad94c71eb7e5987.js:1:236651](http://192.168.0.244:3000/_next/static/chunks/app/page-8ad94c71eb7e5987.js)
XHRPOST
http://192.168.0.244:3000/api/cors/http/192.168.0.244:3900/chatgpt-next-web
[HTTP/1.1 405 Method Not Allowed 56ms]

```js
[WebDav] check 405 Method Not Allowed [page-8ad94c71eb7e5987.js:1:235317](http://192.168.0.244:3000/_next/static/chunks/app/page-8ad94c71eb7e5987.js)
[CORS] target =  /api/cors/http/192.168.0.244:3900/chatgpt-next-web/backup.json [page-8ad94c71eb7e5987.js:1:236651](http://192.168.0.244:3000/_next/static/chunks/app/page-8ad94c71eb7e5987.js)
XHRPOST
http://192.168.0.244:3000/api/cors/http/192.168.0.244:3900/chatgpt-next-web/backup.json
[HTTP/1.1 404 Not Found 50ms]
```

When I click the `Sync button` Console showed this:

```js
[WebDav] get key =  admin 404 Not Found [page-8ad94c71eb7e5987.js:1:235584](http://192.168.0.244:3000/_next/static/chunks/app/page-8ad94c71eb7e5987.js)
[Sync] failed to get remote state SyntaxError: JSON.parse: unexpected end of data at line 1 column 1 of the JSON data
    NextJS 32
[page-8ad94c71eb7e5987.js:1:228903](http://192.168.0.244:3000/_next/static/chunks/app/page-8ad94c71eb7e5987.js)
[CORS] target =  /api/cors/http/192.168.0.244:3900/chatgpt-next-web/backup.json [page-8ad94c71eb7e5987.js:1:236651](http://192.168.0.244:3000/_next/static/chunks/app/page-8ad94c71eb7e5987.js)
[WebDav] set key =  admin 201 Created
```

then Sync is successful and file `backup.json` is created.

**Comment by H0llyW00dzZ** [comment]
> @H0llyW00dzZ Thx for your help. I have tried your version. But it **didn't** worked as well.
> 
> When I clicked the `Check Connection` button, Console showed this:
> 
> ```js
> XHRMKCOL
> [http://192.168.0.244:3000/api/upstash/chatgpt-next-web?endpoint=http://192.168.0.244:3900](http://192.168.0.244:3000/api/upstash/chatgpt-next-web?endpoint=http%3A%2F%2F192.168.0.244%3A3900)
> [HTTP/1.1 404 Not Found 30ms]
> 
> [WebDav] check 404 Not Found [page-1b933c022a7b0acd.js:1:239001](http://192.168.0.244:3000/_next/static/chunks/app/page-1b933c022a7b0acd.js)
> ```
> 
> btw the sysc status icon seems like turned `checked`, that isn't true. ![image](https://private-user-images.githubusercontent.com/41953678/315266704-39e8a6d2-58a6-41cc-af3e-eb97c982448e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTExMTgyODYsIm5iZiI6MTcxMTExNzk4NiwicGF0aCI6Ii80MTk1MzY3OC8zMTUyNjY3MDQtMzllOGE2ZDItNThhNi00MWNjLWFmM2UtZWI5N2M5ODI0NDhlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzIyVDE0MzMwNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWMxNDM5OGFjYmQ0ZTJlMTAzODVkNjNhNjViNmY5MDVmOTA2MDBjMjY1ODIzNmVjMzdhYWJmMzI0ODdkMjY5N2ImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.7rsFBIMPZeoLnmDXqTofQjOHeJ0VPCJPOlU4ydR5Y4E)
> 
> `Sync` button showed this:
> 
> ```js
> XHRGET
> [http://192.168.0.244:3000/api/upstash/chatgpt-next-web/backup.json?endpoint=http://192.168.0.244:3900](http://192.168.0.244:3000/api/upstash/chatgpt-next-web/backup.json?endpoint=http%3A%2F%2F192.168.0.244%3A3900)
> [HTTP/1.1 403 Forbidden 31ms]
> 
> [WebDav] get key =  admin 403 Forbidden [page-1b933c022a7b0acd.js:1:239257](http://192.168.0.244:3000/_next/static/chunks/app/page-1b933c022a7b0acd.js)
> [Sync] failed to get remote state TypeError: t is undefined
>     NextJS 34
> [page-1b933c022a7b0acd.js:1:232439](http://192.168.0.244:3000/_next/static/chunks/app/page-1b933c022a7b0acd.js)
> ```
> 
> Hope those log is useful to your work.
> 
> Thx again.


It seems that the issue is because the `/api/cors/` route was removed, which caused a conflict between the client-side and server-side code.


> I have rolled back to the v2.11.2. When I clicked the Check Connection button, Console showed this:
> 
> ```js
> [CORS] target =  /api/cors/http/192.168.0.244:3900/chatgpt-next-web [page-8ad94c71eb7e5987.js:1:236651](http://192.168.0.244:3000/_next/static/chunks/app/page-8ad94c71eb7e5987.js)
> XHRPOST
> http://192.168.0.244:3000/api/cors/http/192.168.0.244:3900/chatgpt-next-web
> [HTTP/1.1 405 Method Not Allowed 56ms]
> ```
> 
> I have check this request seems like to create a directory(?) by MKCOL method. Normally MKCOL is acceptable.
> 
> When I click the `Sync button` Console showed this:
> 
> ```js
> [WebDav] check 405 Method Not Allowed [page-8ad94c71eb7e5987.js:1:235317](http://192.168.0.244:3000/_next/static/chunks/app/page-8ad94c71eb7e5987.js)
> [CORS] target =  /api/cors/http/192.168.0.244:3900/chatgpt-next-web/backup.json [page-8ad94c71eb7e5987.js:1:236651](http://192.168.0.244:3000/_next/static/chunks/app/page-8ad94c71eb7e5987.js)
> XHRPOST
> http://192.168.0.244:3000/api/cors/http/192.168.0.244:3900/chatgpt-next-web/backup.json
> [HTTP/1.1 404 Not Found 50ms]
> 
> [WebDav] get key =  admin 404 Not Found [page-8ad94c71eb7e5987.js:1:235584](http://192.168.0.244:3000/_next/static/chunks/app/page-8ad94c71eb7e5987.js)
> [Sync] failed to get remote state SyntaxError: JSON.parse: unexpected end of data at line 1 column 1 of the JSON data
>     NextJS 32
> [page-8ad94c71eb7e5987.js:1:228903](http://192.168.0.244:3000/_next/static/chunks/app/page-8ad94c71eb7e5987.js)
> [CORS] target =  /api/cors/http/192.168.0.244:3900/chatgpt-next-web/backup.json [page-8ad94c71eb7e5987.js:1:236651](http://192.168.0.244:3000/_next/static/chunks/app/page-8ad94c71eb7e5987.js)
> [WebDav] set key =  admin 201 Created
> ```
> 
> then Sync is successful and file `backup.json` is created.

For the old version, you may need to temporarily block internal IP addresses because the old version might still be vulnerable to Server-Side Request Forgery (SSRF) when hosted on cloud platforms like AWS due to lack of proper configuration or misconfiguration.

**Comment by H0llyW00dzZ** [comment]
Also, I have already found and identified a better method for handling this issue. However, I plan to start implementing it next month after Eid Mubarak, as I currently don't have much free time available.

**Comment by zhaobudaomingzi** [comment]
> 我在使用坚果云webdav，点击“同步”显示成功了，但是去网盘实际看文件没有更新

我也遇到了这个问题

**Comment by jetaimy** [comment]
使用同样的upstash地址和token，2.10.3可以同步，2.11.3同步失败：
![image](https://github.com/ChatGPTNextWeb/ChatGPT-Next-Web/assets/26423723/2b654369-3294-4e78-8afd-fe6f7ed0f06e)

**Comment by SakuraKoi** [comment]
![image](https://github.com/ChatGPTNextWeb/ChatGPT-Next-Web/assets/19240677/cf359eea-7702-4f05-a63d-4d9d130a0efd)
Cause: Malformed url due to double slash

**Comment by SakuraKoi** [comment]
![image](https://github.com/ChatGPTNextWeb/ChatGPT-Next-Web/assets/19240677/f9eacd19-696a-4449-a576-b26b0deee40d)

![image](https://github.com/ChatGPTNextWeb/ChatGPT-Next-Web/assets/19240677/351309df-4f38-423c-ae5e-c1078429172c)
Cause: `proxyUrl`'s default value in frontend is `/`, then the relative path is not a valid URL
Solution: set `proxyUrl` to your domain

**Comment by SakuraKoi** [comment]
![image](https://github.com/ChatGPTNextWeb/ChatGPT-Next-Web/assets/19240677/9f4c584e-c216-426b-923e-800221582d04)
after fixing(?) the frontend, the backend throwed a similar exception from [here](https://github.com/ChatGPTNextWeb/ChatGPT-Next-Web/blob/main/app/api/webdav/%5B...path%5D/route.ts#L22)
not familiar with Next, guess framework made it a relative path too?

**Comment by SakuraKoi** [comment]
![image](https://github.com/ChatGPTNextWeb/ChatGPT-Next-Web/assets/19240677/cb74e10f-12d5-4ec5-af5e-e9fca223de91)
after set `proxyUrl` to my domain, the backend throwed a similar exception from [here](https://github.com/ChatGPTNextWeb/ChatGPT-Next-Web/blob/main/app/api/webdav/%5B...path%5D/route.ts#L22)
not familiar with Next, guessing it's also a relative path?

**Comment by Dean-YZG** [comment]
> 使用同样的upstash地址和token，2.10.3可以同步，2.11.3同步失败： ![image](https://private-user-images.githubusercontent.com/26423723/319112097-2b654369-3294-4e78-8afd-fe6f7ed0f06e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTMyNTExMTEsIm5iZiI6MTcxMzI1MDgxMSwicGF0aCI6Ii8yNjQyMzcyMy8zMTkxMTIwOTctMmI2NTQzNjktMzI5NC00ZTc4LThhZmQtZmU2ZjdlZDBmMDZlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDE2VDA3MDAxMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQyNTUzZjcxNmQxMzY2NTk2YmY4NTlkMTY5N2JjNmIyMTA0YjQ1MTdiNWYzN2MyMjI3OGY5ZGYxMjhiNmNlNGYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.6xZ7L7xjo6l82_OfKsUUjFubVtwNwOeBh6D8ACnZzOg)

it has been fixed in this pr https://github.com/ChatGPTNextWeb/ChatGPT-Next-Web/pull/4525

**Comment by zhaobudaomingzi** [comment]
> it has been fixed in this pr #4525

还是同步失败

**Comment by mghcool** [comment]
到底应该怎样做才能同步成功？现在2.12.3版本还是不能同步
